src/main/java/net/masterthought/cucumber/json/Embedding.java

```diff
@@ -96,6 +104,19 @@ public String getExtension() {
             case "application/vnd.ms-excel":
                 return "xls";
             default:
+                // assert the name is file-name formatted --> try file-name extension
+                if (name != null && name.contains(".")) {
+                    String extension = name.substring(name.lastIndexOf('.') + 1);
+                    // the extension might by usable
+                    if (extension.matches("[a-z0-9]+")) return extension;
```

Comment:
would it match "www.google.com/gmail ?

Reply:
Dear Damian, at this point we are not referring to the MIME type, but to the name instead. Your example will match the if-condition. However, the extracted 'extension' will not match the file-extension pattern:
```
jshell> String name = "www.google.com/gmail";
name ==> "www.google.com/gmail"

jshell> name.contains(".")
$5 ==> true

jshell> String extension = name.substring(name.lastIndexOf('.') + 1);
extension ==> "com/gmail"

jshell> extension.matches("[a-z0-9]+")
$7 ==> false
```

Reply:
I'd like to add, that if your example would be the MIME type, the derived extension would be "gmail".
However, this would be an invalid MIME-type because a MIME-type's type cannot contain a ".".
Only the subtype might be written as a tree. (An example is "application/vnd.ms-excel".)